Formula/go-boring.rb

```diff
@@ -0,0 +1,80 @@
+class GoBoring < Formula
+  desc "Go programming language with BoringCrypto"
+  homepage "https://go.googlesource.com/go/+/dev.boringcrypto/README.boringcrypto.md"
+  url "https://go-boringcrypto.storage.googleapis.com/go1.16.2b7.src.tar.gz"
+  version "1.16.2b7"
+  sha256 "8991c28e30bdd9b34076c2871ab69aac7f306dba08d202aa7301f3adb54ad9f3"
+  license "BSD-3-Clause"
+
+  livecheck do
+    url "https://go-boringcrypto.storage.googleapis.com/"
+    regex(/go(\d+(?:\.\d+)+b\d+)[._-]src\.t/i)
```

Comment:
```suggestion
    regex(/>go[._-]?v?(\d+(?:\.\d+)+(?:b\d+)?)[._-]src\.t/i)
```

This is functional but it could benefit from a few small tweaks:

* Adding `>` to the start of the regex will restrict the beginning of the filename, ensuring we only match filenames starting with `go` (i.e., not anything that ends with `go` like `something-else-go`)
* Though the filenames use a `go1.2.3` format, adding an optional `[._-]?` between the software name and version allows the regex to continue to match if the filename format changes to start with something like `go-`, `go.`, etc. in the future
* Making the `b\d+` part of the version optional allows us to also match a version like `1.2.3` instead of only `1.2.3b4`, if this format ever appears in the future

Reply:
> * Making the `b\d+` part of the version optional allows us to also match a version like `1.2.3` instead of only `1.2.3b4`, if this format ever appears in the future

The `b\d+` part isn't optional, I think. It's how they indicate the version of the BoringCrypto pieces of their branch.

Reply:
I had a feeling that may be the case. I've removed that part of the suggestion and we can always add it later as needed.